File: python3/max_consecutive_ones.py
# leetcode

class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        ans, dp = 0, 0
        
        for i in nums:
            if i == 1:
                ans += 1
                dp = max(dp, ans)
            else:
                ans = 0
            
        return dp

File: golang/max_consecutive_ones.go
package main

// Space	: O(1)
// Time		: O(n)

func findMaxConsecutiveOnes(nums []int) int {
	ans, count := 0, 0
	l := len(nums)

	for i := 0; i < l; i++ {
		if nums[i] == 0 {
			ans = Max(ans, count)
			count = 0
		} else {
			count++
		}
	}

	return Max(ans, count)
}
